pkg/devicetemplates/ttscsv.go

```diff
@@ -371,6 +376,9 @@ func (t *ttsCSV) Convert(ctx context.Context, r io.Reader, ch chan<- *ttnpb.EndD
 		if c, ok := determineComma(head); ok {
 			comma = c
 		}
+		if ok := isBOM(head); ok {
```

Comment:
Could we instead peek the first rune (`buf.ReadRune()`) after we initialize `buf`, and then compare it with `'\uFEFF'` ? That's a bit more idiomatic than byte comparison and handling.

If the first rune is not `'\uFEFF'`, we can use `buf.UnreadRune()`.